src/signalhandler.cc

```diff
@@ -216,8 +220,13 @@ void DumpSignalInfo(int signal_number, siginfo_t* siginfo) {
   std::ostringstream oss;
   oss << std::showbase << std::hex << std::this_thread::get_id();
   formatter.AppendString(oss.str().c_str());
-
+#if defined (HAVE_SYS_SYSCALL_H) && defined (HAVE_SYS_TYPES_H)
+  pid_t tid = syscall(SYS_gettid);
```

Comment:
Unfortunately, using `syscall` triggers the following error on macOS caused by a deprecation warning:
```console
src/signalhandler.cc:224:15: error: 'syscall' is deprecated: first deprecated in macOS 10.12 - syscall(2) is unsupported; please switch to a supported interface. For SYS_kdebug_trace use kdebug_signpost(). [-Werror,-Wdeprecated-declarations]
  pid_t tid = syscall(SYS_gettid);
```
We try to avoid deprecated APIs.

Reply:
change to: print LWP id only on Linux platforms